File: app/src/main/java/com/example/kocja/rabbiter_online/services/alertIfNotAlertedService.java
package com.example.kocja.rabbiter_online.services;

import android.app.IntentService;
import android.content.Intent;
import android.util.Log;

import com.example.kocja.rabbiter_online.managers.GsonManager;
import com.example.kocja.rabbiter_online.managers.HttpManager;
import com.example.kocja.rabbiter_online.databases.Events;

import androidx.annotation.Nullable;

/**
 * Created by kocja on 28/02/2018.
 */

public class alertIfNotAlertedService extends IntentService {
    public alertIfNotAlertedService(){
        super("This is alertIfNotAlertedService");
    }
    @Override
    protected void onHandleIntent(@Nullable Intent intent) {
        HttpManager.INSTANCE.getRequest("seekNotAlertedEvents", response -> {
            for(Events event : GsonManager.INSTANCE.getGson().fromJson(response,Events[].class)){
                Log.v("Oops","This guy was not started");
                NotifyUser.schedule(this,event.getDateOfEventMilis(),event.getEventUUID().toString());
            }
        });
        /*
        SQLite.select()
                .from(Events.class)
                .where(Events_Table.notificationState.eq(Events.NOT_YET_ALERTED))
                .async()
                .queryListResultCallback((transaction, tResult) -> {
                }).execute();
          */
    }
}

File: app/src/main/java/com/example/kocja/rabbiter_online/services/AlertIfNotAlertedService.kt
package com.example.kocja.rabbiter_online.services

import android.app.IntentService
import android.content.Intent
import android.util.Log
import androidx.core.app.JobIntentService
import com.example.kocja.rabbiter_online.managers.WebService
import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.launch
import org.koin.android.ext.android.inject


/**
 * Created by kocja on 28/02/2018.
 */

class AlertIfNotAlertedService : IntentService("This is AlertIfNotAlertedService") {
    private val fetcher : WebService by inject()
    override fun onHandleIntent(intent: Intent?) {

        CoroutineScope(Dispatchers.IO).launch{
            val result = fetcher.findNotAlertedEvents()
            result.forEach{
                Log.v("Oops", "This guy was not started")
                //NotifyUser.schedule(this, event.dateOfEventMilis, event)
                EventTriggered.scheduleWorkManager(this@AlertIfNotAlertedService,it.dateOfEventMilis,it.eventUUID)
            }
        }
    }
}

